File: Backend/Microservices/Commodity/Eefa.Commodity.Application/Commands/CommodityCategoryPropertyItem/Update/UpdateCommodityCategoryPropertyItemCommand.cs
using AutoMapper;
using Eefa.Common.CommandQuery;
using Eefa.Common;
using Eefa.Common.Data;
using MediatR;
using System.Threading;
using System.Threading.Tasks;

namespace Eefa.Commodity.Application.Commands.CommodityCategoryPropertyItem.Update
{
    public class UpdateCommodityCategoryPropertyItemCommand : CommandBase, IRequest<ServiceResult>, IMapFrom<Data.Entities.CommodityCategoryPropertyItem>, ICommand
    {
        public int Id { get; set; }


        /// <summary>
        /// کد ویژگی گروه
        /// </summary>
        public int CategoryPropertyId { get; set; } = default!;

        /// <summary>
        /// کد والد
        /// </summary>
        public int? ParentId { get; set; }

        /// <summary>
        /// عنوان
        /// </summary>
        public string Title { get; set; } = default!;

        /// <summary>
        /// نام اختصاصی
        /// </summary>
        public string UniqueName { get; set; } = default!;

        /// <summary>
        /// کد
        /// </summary>
        public string? Code { get; set; }

        /// <summary>
        /// ترتیب نمایش
        /// </summary>
        public int OrderIndex { get; set; } = default!;

        /// <summary>
        /// فعال است؟
        /// </summary>
        public bool IsActive { get; set; } = default!;

        public void Mapping(Profile profile)
        {
            profile.CreateMap<UpdateCommodityCategoryPropertyItemCommand, Eefa.Commodity.Data.Entities.CommodityCategoryPropertyItem>()
                .IgnoreAllNonExisting();
        }
    }


    public class UpdateCommodityCategoryPropertyItemCommandHandler : IRequestHandler<UpdateCommodityCategoryPropertyItemCommand, ServiceResult>
    {
        private readonly IRepository<Data.Entities.CommodityCategoryPropertyItem> _repository;
        private readonly IMapper _mapper;

        public UpdateCommodityCategoryPropertyItemCommandHandler(IRepository<Data.Entities.CommodityCategoryPropertyItem> repository, IMapper mapper)
        {
            _mapper = mapper;
            _repository = repository;
        }

        public async Task<ServiceResult> Handle(UpdateCommodityCategoryPropertyItemCommand request, CancellationToken cancellationToken)
        {
            var entity = await _repository.Find(request.Id);

            _mapper.Map<UpdateCommodityCategoryPropertyItemCommand, Data.Entities.CommodityCategoryPropertyItem>(request, entity);
            _repository.Update(entity);
            await _repository.SaveChangesAsync(cancellationToken);

            return ServiceResult.Success();
        }
    }
}

File: Frontend/angular-app/src/app/modules/commodity/repositories/commodity-category-property-item/commands/update-commodity-category-property-item-command.ts
import {Inject} from "@angular/core";
import {ServiceResult} from "../../../../../core/models/service-result";
import {IRequest, IRequestHandler} from "../../../../../core/services/mediator/interfaces";
import {HttpService} from "../../../../../core/services/http/http.service";
import {ApplicationError} from "../../../../../core/exceptions/application-error";
import {MediatorHandler} from "../../../../../core/services/mediator/decorator";
import {NotificationService} from "../../../../../shared/services/notification/notification.service";
import {CommodityCategoryPropertyItem} from "../../../entities/commodity-category-property-item";
import {ValidationRule} from "../../../../../core/validation/validation-rule";
import {HttpRequest} from "../../../../../core/services/http/http-request";
import {EntityStates} from "../../../../../core/enums/entity-states";

export class UpdateCommodityCategoryPropertyItemCommand extends IRequest<UpdateCommodityCategoryPropertyItemCommand, CommodityCategoryPropertyItem> {
  public id: number | undefined = undefined;
  public categoryPropertyId: number | undefined = undefined;
  public parentId: number | undefined = undefined;
  public title: string | undefined = undefined;
  public uniqueName: string | undefined = undefined;
  public code: string | undefined = undefined;
  public orderIndex: number | undefined = undefined;
  public isActive: boolean | undefined = undefined;
  constructor() {
    super();
  }

  mapFrom(entity: CommodityCategoryPropertyItem): UpdateCommodityCategoryPropertyItemCommand {
    this.mapBasics(entity,this)
    return this;
  }

  mapTo(): CommodityCategoryPropertyItem {
    throw new ApplicationError(UpdateCommodityCategoryPropertyItemCommand.name, this.mapTo.name, 'Not Implemented Yet')
  }

  get url(): string {
    return "/commodity/commodityCategoryPropertyItem/update";
  }

  get validationRules(): ValidationRule[] {
    return [];
  }
}

@MediatorHandler(UpdateCommodityCategoryPropertyItemCommandHandler.name)
export class UpdateCommodityCategoryPropertyItemCommandHandler implements IRequestHandler<UpdateCommodityCategoryPropertyItemCommand, CommodityCategoryPropertyItem> {
  constructor(
    @Inject(HttpService) private _httpService: HttpService,
    @Inject(NotificationService) private _notificationService: NotificationService,
  ) {
  }

  async Handle(request: UpdateCommodityCategoryPropertyItemCommand): Promise<CommodityCategoryPropertyItem> {
    let httpRequest: HttpRequest<UpdateCommodityCategoryPropertyItemCommand> = new HttpRequest<UpdateCommodityCategoryPropertyItemCommand>(request.url, request);
    // httpRequest.Query  += ;
    // httpRequest.Headers = httpRequest.Headers.append()

    return await this._httpService.Put<UpdateCommodityCategoryPropertyItemCommand, ServiceResult<CommodityCategoryPropertyItem>>(httpRequest).toPromise().then(response => {
      // this._notificationService.showHttpSuccessMessage()
      return response.objResult
    })
  }
}
